crypt.go

```diff
@@ -127,6 +129,105 @@ type StandardEncryptionVerifier struct {
 	EncryptedVerifierHash []byte
 }
 
+// hashAlgorithmName
+type hashAlgorithmName = string
+
+var hashAlgorithmSHA512 hashAlgorithmName = "SHA-512"
+
+type saltHashData struct {
+	AlgorithmName hashAlgorithmName
+	Password      string
+	SpinCount     int // default 100000 1e5
+}
+type saltHashResult struct {
+	AlgorithmName hashAlgorithmName
+	HashValue     string
+	// An array of bytes that specifies the salt value used during password hash generation.
+	// When computing hashes for new passwords, this MUST be generated using an arbitrary
+	// pseudorandom function. When verifying a password, the salt value retrieved from the
+	// document MUST be used. The salt MUST NOT be larger than 65,536 bytes.
+	SaltValue string
+	// default: 100000 1e5. A 32-bit unsigned integer that specifies the number of
+	// times to iterate on a hash of a password. It MUST NOT be greater than 10,000,000.
+	SpinCount int
+}
+
+/*
+genISOPasswordHash implements "ISO password hashing algorithm"
```

Comment:
Just code style suggestions: using the `//` comment delimiter comment out document block.